blockchain/indexers/addrindex.go

```diff
@@ -809,6 +809,16 @@ func (idx *AddrIndex) DisconnectBlock(dbTx database.Tx, block *btcutil.Block,
 	return nil
 }
 
+// PruneBlock is invoked when an older block is deleted after it's been
+// processed.
+// NOTE: For AddrIndex, it's a no-op as AddrIndex isn't allowed to be enabled
```

Comment:
Wouldn't we want to _reclaim_ the space on disk due to the addr index? Or is the idea that you need to first drop the entire thing, _then_ enable pruning?

Reply:
Currently if you have had the addr index enabled, it won't prune the ones you've already indexed unless you do it manually.

I guess we could do either one of these:

1: Don't do anything with the already present addr index data (current)
2: Drop the addr index if pruning is enabled
3: Force the user to drop the addr index before allowing them to enable pruning

I'd favor either 1 or 3. Explicit user action or doing nothing seems better.

Reply:
> 3: Force the user to drop the addr index before allowing them to enable pruning

I prefer this path myself. As otherwise, the queries of indexes like the addr index wouldn't be possible as it wants to return the block regions where the transaction exist, but they may not be accessible after pruning. 

The only index that can still serve its purpose with pruned data is the neutrino filter index, as querying it doesn't require any block data at all (just slinging out the premade filters for each block).